pkg/api/client/client.go

```diff
@@ -0,0 +1,111 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/bmc-toolbox/common"
+	rivets "github.com/metal-toolbox/rivets/types"
+)
+
+const (
+	versionEndpoint            = "/api/version"
+	componentsEndpoint         = "/components"
+	inbandInventoryEndpoint    = "/inventory/in-band"
+	outofbandInventoryEndpoint = "/inventory/out-of-band"
+)
+
+type ServerComponents map[string][]*rivets.Component
+
+// Client can perform queries against the Component Inventory Service.
+type Client interface {
```

Comment:
+1. Consider adding a mock implementation (in a sub-folder) with `github.com/uber-go/gomock`.